pkg/manager/member/pd_member_manager.go

```diff
@@ -268,6 +271,15 @@ func (pmm *pdMemberManager) syncTidbClusterStatus(tc *v1alpha1.TidbCluster, set
 	healthInfo, err := pdClient.GetHealth()
 	if err != nil {
 		tc.Status.PD.Synced = false
+		// get endpoints info
```

Comment:
`GetCluster()` will also be failed if there are no endpoints.

so we should move these codes to https://github.com/pingcap/tidb-operator/pull/545/files#diff-006f391cd7cdae269e89bd77e21c1a6fR266

Reply:
Hi, I think that it may be better to check health state at first, and check endpoints' status when it's unhealthy, just like the previous code. Here is what I think:
- It makes no difference to the whole component whether we check health state first or not. 
- Unit tests based on `GetHealth` rather than `GetCluser` are easier to understand, such as: https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pd_member_manager_test.go#L209. And current unit tests on endpoints are based on whether `GetHealth` fail  or not. 
- More intuitive.

Reply:
OK, i am fine with both orders. But these codes must be moved to the first method.

Reply:
Thank you for your patience！